src/commands/ban.ts

```diff
@@ -0,0 +1,70 @@
+import { PermissionFlagsBits, SlashCommandBuilder } from 'discord.js';
+import { CmdFileArgs } from '../typings/Extensions.js';
+
+export const ephemeral = true;
+export const data = new SlashCommandBuilder()
+  .setName('ban')
+  .setDescription('Permanently bans a user from the server')
+  .addUserOption((option) => {
+    return option.setName('user').setDescription('The user to ban').setRequired(true);
+  })
+  .addStringOption((option) => {
+    return option.setName('reason').setDescription('Reason for ban, if any').setAutocomplete(true).setRequired(false);
+  })
+  .addNumberOption((option) => {
+    return option.setName('delete_messages').setDescription('Time to delete messages').setChoices(
+      {
+        name: "Don't Delete Any",
+        value: 0
+      },
+      {
+        name: '1 Hour',
+        value: 3600
+      },
+      {
+        name: '6 Hours',
+        value: 21600
+      },
+      {
+        name: '12 Hours',
+        value: 43200
+      },
+      {
+        name: '24 Hours',
+        value: 86400
+      },
+      {
+        name: '3 Days',
+        value: 259200
+      },
+      {
+        name: '7 Days',
+        value: 604800
+      }
+    );
+  })
```

Comment:
Are we able to make the type in how many hours or days they can delete? Like 2d5h13m

Reply:
We could yes. I think I made a parseTime function before. This would just need to be converted to a string option